da/celestia/celestia_test.go

```diff
@@ -10,9 +10,34 @@ import (
 )
 
 func TestDataRequestErrorToStatus(t *testing.T) {
+	randErr := errors.New("some random error")
+	var test = []struct {
+		statusCode da.StatusCode
+		err        error
+	}{
+		// Status Success Cases
+		{da.StatusSuccess, nil},
+		{da.StatusSuccess, da.ErrNamespaceNotFound},
+		{da.StatusSuccess, errors.Join(randErr, da.ErrNamespaceNotFound, randErr)},
+
+		// TODO: cases that need investigating, are these possible? If
+		// so, is this the correct status code?
+		{da.StatusSuccess, errors.Join(da.ErrEDSNotFound, da.ErrNamespaceNotFound)},
```

Comment:
according to celestia-node, these errors are not joined.

Reply:
i think it is still a valid test case because it is verifying which error gets priority in the switch statement.

If that logic changes, then this test catches that now. 

Additionally if there is a change in node, that somehow allows for these errors to be combined then it will catch it. 
Which also catches bugs in how rollkit is checking the errors because in the method we are combining errors. So we might change something that checks errors outside of this method.